### ISSUE TITLE
could you please provide your pre-trained model?

### ISSUE THREAD

**Issue by Colin97** [lynetcha/completion3d#8]
I followed the instruction to train the topnet for 300 epochs using the data provided.
However, the results don't seem as good as the paper presents.
Could you please provide your pre-trained model?

**Comment by Colin97** [comment]
According to my experiments, AtlasNet and PCN have much lower Chamfer distance than your TopNet.

**Comment by Colin97** [comment]
In PCN's code, the chamfer distance is calculated as:

def chamfer(pcd1, pcd2):
    dist1, _, dist2, _ = tf_nndistance.nn_distance(pcd1, pcd2)
    dist1 = tf.reduce_mean(tf.sqrt(dist1))
    dist2 = tf.reduce_mean(tf.sqrt(dist2))
    return (dist1 + dist2) / 2

According to PCN's paper and my experiments, PCN's chamfer distance should be about 0.01, while in your experiment it becomes about 0.02~0.03 (translated, since chamfer distance is calculated in a different way in your code).

I cannot reproduce other results as well.

**Comment by sheshappanavar** [comment]
@Colin97 did you train using the PyTorch implementation or the TensorFlow one?
Also, can you please share the results which you got from the training? 

Thanks very much

**Comment by Colin97** [comment]
@sheshappanavar  I used the TensorFlow version for TopNet.
Here are the results_val_300:
```
#ParametersTotal 9965117
#ParametersEncoder 821504
#ParametersDecoder 9143613
plane Generator_emd 0.145209
plane Generator_dist 0.000808
cabinet Generator_emd 0.152978
cabinet Generator_dist 0.002192
car Generator_emd 0.087199
car Generator_dist 0.001143
chair Generator_emd 0.186171
chair Generator_dist 0.002337
lamp Generator_emd 0.249861
lamp Generator_dist 0.001748
couch Generator_emd 0.153754
couch Generator_dist 0.001918
table Generator_emd 0.151672
table Generator_dist 0.001969
watercraft Generator_emd 0.142658
watercraft Generator_dist 0.001177
Generator Class Mean EMD 0.158688
Generator Class Mean DIST 0.001661
```

**Comment by xiazhiwang** [comment]
@Colin97 Hi, may I know how do you get the ground truth for test data? I only found partial input for test data. Thanks a lot.

**Comment by lynetcha** [comment]
Apologies for the delay in response. The benchmark website is back on and you can submit your trained model for evaluation on the test set. I am looking at the result differences for the Tensorflow models(Folding, PCN, Tensorflow) and I believe they are related to the commit I just made. I will reproduce the experiments and get back to you about the results. In the meantime, one of the pretrained TopNet models has been uploaded on the benchmark for comparison.
Concerning the loss - we use the original definition of the chamfer loss as defined [here ](https://arxiv.org/abs/1612.00603).

**Comment by doublemanyu** [comment]
@Colin97 hi, i have trained the model, but i don't konw how to test. Should I run the shapenet.py under the ./shared/datasets/? Cause I found the test_propocess function.

**Comment by slothfulxtx** [comment]
I have trained almost all model on completion3d dataset, using the scripts provided in this repo. However, I can't reproduce the perfomance of most of them on the leaderboard. Thus, could you please provide your pre-trained model? Otherwise I cannot do further research on completion3d dataset. Here are part of my results


Method | CD(10-4) | Airplane | Cabinet | Car | Chair | Lamp | Sofa | Table | Watercraft
-- | -- | -- | -- | -- | -- | -- | -- | -- | --
topnet | 16.84 | 10.11 | 21.7 | 13.21 | 23.77 | 18.46 | 20.52 | 15.62 | 10.72
pointnetfcae official | 21.49 | 14.34 | 23.91 | 14.12 | 25.55 | 26.86 | 25.08 | 26.23 | 15.15
atlas net official | 20.6 | 12.62 | 26.91 | 14.59 | 27.92 | 24.88 | 23.17 | 20.17 | 13.79

@lynetcha @vineetsk1 @micaeltchapmi
